test_scripts/Defects/5_1/1004_App_is_disconnected_due_to_PROTOCOL_VIOLATION_when_start_audio_streaming_after_rejected_2_times_then_accepted.lua

```diff
@@ -0,0 +1,81 @@
+---------------------------------------------------------------------------------------------------
+-- User story: https://github.com/smartdevicelink/sdl_core/issues/1004
+-- Description: App is disconnected due to PROTOCOL_VIOLATION when start audio streaming after rejected 2 times then accepted
+-- Precondition:
+-- 1) SDL and HMI are started.
+-- 2) App is registered and activated
+-- In case:
+-- 1) Press "Audio Service - Show" button
+-- 2) Press "Start Service"
+-- 3) Press "Cancel" on pop-up
+-- 4) Press "Cancel" on pop-up
+-- 5) Press "OK" pop-up
+-- 6) Press "Start File Streaming"
+-- Expected result:
+-- 1) Audio Streaming is started.
+-- Actual result:
+-- App is unregistered with reason "PROTOCOL_VIOLATION".
+---------------------------------------------------------------------------------------------------
+--[[ Required Shared libraries ]]
+local runner = require('user_modules/script_runner')
+local common = require('test_scripts/SDL5_0/TTSChunks/common')
+local events = require('events')
+local constants = require("protocol_handler/ford_protocol_constants")
+
+--[[ Test Configuration ]]
+runner.testSettings.isSelfIncluded = false
+config.defaultProtocolVersion = 3
+config.application1.registerAppInterfaceParams.appHMIType = { "NAVIGATION" }
+
+--[[ Local Constants ]]
+local delay = 800
+
+--[[ Local Functions ]]
+local function expectEndService(pServiceId)
+  local event = events.Event()
+  event.matches = function(_, data)
+    return data.frameType == constants.FRAME_TYPE.CONTROL_FRAME
+    and data.serviceType == pServiceId
+    and data.sessionId == common.getMobileSession().mobile_session_impl.control_services.session.sessionId.get()
+    and data.frameInfo == constants.FRAME_INFO.END_SERVICE
+  end
+  return common.getMobileSession():ExpectEvent(event, "EndService")
+end
+
+local function startStreaming()
+  expectEndService(10)
+  :Times(0)
+  common.getHMIConnection():ExpectRequest("Navigation.StartAudioStream")
+  :Do(function(exp, data)
+    local function responseRejected()
+      common.getHMIConnection():SendError(data.id, data.method, "REJECTED", "Ignored by USER")
+    end
+    local function responseSuccess()
+      common.getHMIConnection():SendResponse(data.id, data.method, "SUCCESS", {})
+    end
+    if exp.occurences >= 1 and exp.occurences <= 3 then
+      RUN_AFTER(responseRejected, delay)
+    elseif exp.occurences == 4 then
+      RUN_AFTER(responseSuccess, delay)
+    end
+  end)
+  :Times(AnyNumber())
```

Comment:
Please expect the exact number of massages.

Reply:
@GetmanetsIrina You can find the updates in the commit 05f7cf9